src/psyclone/domain/gocean/transformations/gocean_opencl_trans.py

```diff
@@ -0,0 +1,157 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2021, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors R. W. Ford, A. R. Porter and S. Siso, STFC Daresbury Lab
+
+'''This module contains the GOcean-specific OpenCL transformation.
+'''
+
+import six
+from psyclone.psyGen import Transformation
+from psyclone.undoredo import Memento
+from psyclone.transformations import TransformationError, KernelTrans
+from psyclone.psyGen import args_filter, InvokeSchedule
+from psyclone.gocean1p0 import GOInvokeSchedule
+
+
+class GOOpenCLTrans(Transformation):
+    '''
+    Switches on/off the generation of an OpenCL PSy layer for a given
+    InvokeSchedule. Additionally, it will generate OpenCL kernels for
+    each of the kernels referenced by the Invoke. For example:
+
+    >>> ocl_trans = GOOpenCLTrans()
```

Comment:
Is it possible to make this a fully-working example (#1220)?

Reply:
Done (I looked at other examples and used the same pattern - although in the future it will be good to find a way to simplify all the extra lines that are not that relevant to this snippet and improve the path location)

Reply:
Thanks, and yes.